### PR TITLE
feat(transformer/typescript): support for transform TSImportEqualsDeclaration

### PR DESCRIPTION
**Pull request by Dunqing**
---

### CODE REVIEW

**Comment by Dunqing** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/2998?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#2998** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/2998?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#2997** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/2997?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @Dunqing and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: crates/oxc_transformer/src/typescript/annotations.rs [hunk 1]
```diff
@@ -543,13 +543,73 @@ impl<'a> TypeScriptAnnotations<'a> {
...
+                binding,
+                Some(init),
+                false,
+            ))
+        };
+        let variable_declaration =
+            self.ctx.ast.variable_declaration(SPAN, kind, decls, Modifiers::empty());
+
+        Declaration::VariableDeclaration(variable_declaration)
+    }
```
Comment (Boshen) [inline crates/oxc_transformer/src/typescript/annotations.rs:588]: Should this be in this file? This is TypeScript transform so move this to `typescript.rs`?

Comment (Dunqing) [reply crates/oxc_transformer/src/typescript/annotations.rs]: I moved this to `module.rs`

**Comment by Boshen** [comment]
Oops I didn't use graphite to merge.

**Review by Boshen — APPROVED** [review]

**Pull request merged**